### PR TITLE
Added basic Aquastream XT Hardware Support

### PR DESCRIPTION
**Pull request by Aeronaut**
I have added support for aquacomputer's Aquastream XT water pump.
TODO:
  * Check tested and fix unknown variables in AquastremXT::Update()
  * Check if property AquastremXT::Variant is valid interpreted
  * Implement Fan Control in AquastremXT::SetControl()

I will add more Hardware but need donations for buying them.

Credits also to mikeobrien for his HidLibrary https://github.com/mikeobrien/HidLibrary
# Donation

[Donate](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=EWZUDYRS53BFC)

### CODE REVIEW

**Comment by Phyxion** [comment]
I'll review this later, but one point of feedback I have for now. Can you change the HidLibrary to use NuGet? No point in having to copy the sources manually when it can be done automatically.

**Comment by Aeronaut** [comment]
I tried to do so, but it gives an error:
Das Paket "hidlibrary 3.2.46" wurde nicht mit dem Projektzielframework ".NETStandard,Version=v2.0", sondern mit ".NETFramework,Version=v4.6.1" wiederhergestellt. Dieses Paket ist mit Ihrem Projekt möglicherweise nicht vollständig kompatibel.
Compiling gives CS1061"HidDevice" does not contain a definition for "ProductName" and no ProductName-extension method accepting a first argument of type "HidDevice" could be found (are you missing a using directive or an assembly reference?)

**Comment by Phyxion** [comment]
Took a while to review, sorry. The library doesn't seem to support .NET Standard indeed.  This is then OK I think until the author decides to change that. I'll leave some comments (there's just a few really).

Do you also want to do the points in TODO before this gets merged?

File: Hardware/Aquacomputer/AquastreamXT.cs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            }
+        }
+
+        public override void Close()
+        {
+            device.CloseDevice();
+            base.Close();
+        }
+
+        //TODO: Check tested and fix unknown variables
```
Comment (Phyxion) [inline Hardware/Aquacomputer/AquastreamXT.cs]: ^

File: Hardware/Aquacomputer/AquastreamXT.cs [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+            ActivateSensor(frequencys[1]);            
+        }
+
+        //TODO: Implement Fan Control
+        private void SetControl(byte? v)
+        {
+            throw new NotImplementedException();
+        }
+
+        //TODO: Check if valid
```
Comment (Phyxion) [inline Hardware/Aquacomputer/AquastreamXT.cs]: ^

File: Hardware/Aquacomputer/AquastreamXT.cs [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+            fanControl.Control = control;
+            ActivateSensor(fanControl);
+
+            frequencys[0] = new Sensor("Pump Frequency", 0, SensorType.Frequency, this, new ParameterDescription[0], settings);
+            ActivateSensor(frequencys[0]);
+            frequencys[1] = new Sensor("Pump MaxFrequency", 1, SensorType.Frequency, this, new ParameterDescription[0], settings);
+            ActivateSensor(frequencys[1]);            
+        }
+
+        //TODO: Implement Fan Control
```
Comment (Phyxion) [inline Hardware/Aquacomputer/AquastreamXT.cs]: ^

**Comment by Aeronaut** [comment]
Reporting back again...
New Patch: https://github.com/Aeronaut/LibreHardwareMonitor/commit/e2221d9208317833b01df94d0f828c2f59ceef4f.patch

As for the TODOs, I will have a look on "External Fan RPM", but this can take a few weeks or month...
I will ask for the code to unlook the "Internal Pump Flow Sensor" or for calculation values for this...
So feel free to merge, or wait until I have more Infos, but I think its quite completed now for most common use.

**Comment by Aeronaut** [comment]
Like that? Sorry im new to Github and this kind of tasks are still uncommon for me ;-)

File: Hardware/Aquacomputer/AquastreamXT.cs [hunk 4]
```diff
@@ -111,13 +130,13 @@ public AquastreamXT(HidDevice dev, ISettings settings) : base("Aquastream XT", n
...
-            fanControl.Control = control;
-            ActivateSensor(fanControl);
+            _fanControl.Control = control;
+            ActivateSensor(_fanControl);
 
-            frequencys[0] = new Sensor("Pump Frequency", 0, SensorType.Frequency, this, new ParameterDescription[0], settings);
-            ActivateSensor(frequencys[0]);
-            frequencys[1] = new Sensor("Pump MaxFrequency", 1, SensorType.Frequency, this, new ParameterDescription[0], settings);
-            ActivateSensor(frequencys[1]);            
+            _frequencys[0] = new Sensor("Pump Frequency", 0, SensorType.Frequency, this, new ParameterDescription[0], settings);
```
Comment (Phyxion) [inline Hardware/Aquacomputer/AquastreamXT.cs]: _frequencies is the name 👍

**Comment by Phyxion** [comment]
Yes, much better, so you managed to get HidLibrary working with NuGet? In that case you can also remove the code from the commit (External/HidLibrary/...). I can squash and merge it then, rest looks all good (and left one tiny thing).

**Comment by Aeronaut** [comment]
OK, done...  HidLibrary working with NuGet: It shows a Warning but compiles...

**Comment by Phyxion** [comment]
I'll try it out later and see if I can get it to compile here as well. If that's working then it should be OK 👍

**Comment by Phyxion** [comment]
Sorry for taking so long to try this out, It's indeed working, although there's a warning in VS. Given the hid lib code works just fine in old .NET versions I think we can get away with keeping this warning. I did remove the debug stuff as the debug SensorType wasn't present.

**Pull request closed without merging**